src/StringIntersection.java

```diff
@@ -36,6 +37,69 @@ public static String intersection(String str1, String str2) {
         return output;
 	}
 	
+	
+	
+	
+	/**
+	 * method that takes two strings and returns the intersection,
+	 * which is the string of characters that exist in both strings
+	 * order is not preserved
+	 * 
+	 * This is accomplished with a hash map.
+	 * HashMap<Character, Integer>
+	 * first put a count of all chars in str1 into HashMap
+	 * loop thru str2 and if get(c) > 0 
+	 * then output.append and replace(Int--)
+	 *  
+	 */
+	public static String intersectionHM(String str1, String str2) {
+
+
+		// create hashmap to hold Character => Integer 
+		HashMap<Character, Integer> hm = new HashMap<Character, Integer>();
+
+		Character c;
+		String output = "";
+
+		for(int i = 0; i < str1.length(); i++) {
+			c = new Character(str1.charAt(i));
+			/*
+			 * loop through str1 to create or increment the
+			 * Integer value for the count of chars c  in str1
+			 * if a char exists in str1 one or more times,
+			 * the Integer value will be the count of chars
+			 * 
+			 */
+			if(hm.containsKey(c))
```

Comment:
while syntactically correct, i would encourage you to always use { } to wrap conditional blocks (even if it's one line). it's a good practice to get in to, as it makes it easier for other developers to read the code, but also debugging for you.